docs/develop/typescript/set-up.mdx

```diff
@@ -0,0 +1,458 @@
+
+const { activity } = proxyActivities<typeof activityTypes>({
+startToCloseTimeout: '10 seconds',
+});
+
+export async function sayHelloWorkflow(name: string): Promise<string> {
+const response = await activity(name);
+return response;
+}
+`}
+
+</CodeSnippet>
+</>
+}>
+
+### Create the Workflow
+
+Workflows orchestrate Activities and contain the application logic. Temporal Workflows are resilient. They can run—and keeping running—for years, even if the underlying infrastructure fails. If the application itself crashes, Temporal will automatically recreate its pre-failure state so it can continue right where it left off.
+
+Create a file called `workflows.ts`:
+
+</SetupStep>
+
+<SetupStep code={
+<>
+
+<p><strong>Create a Worker file (worker.ts):</strong></p>
+<CodeSnippet language="typescript">
+{`import { Worker } from '@temporalio/worker';
+import * as activitiy from './activity';
+
+async function run() {
+// Step 1: Register Workflows and Activities with the Worker and connect to
+// the Temporal server.
+const worker = await Worker.create({
+workflowsPath: require.resolve('./workflows'),
+activities: activitiy,
+taskQueue: 'my-task-queue',
+});
+// Worker connects to localhost by default and uses console.error for logging.
+// Customize the Worker by passing more options to create():
+// https://typescript.temporal.io/api/classes/worker.Worker
+// If you need to configure server connection parameters, see docs:
+// https://docs.temporal.io/typescript/security#encryption-in-transit-with-mtls
+
+// Step 2: Start accepting tasks on the my-task-queue queue
+await worker.run();
+}
+
+run().catch((err) => {
+console.error(err);
+process.exit(1);
+});
+`}
+
+</CodeSnippet>
+
+    <p><strong>Run the Worker:</strong></p>
+    <CodeSnippet language="bash">
+    npm run start
+    </CodeSnippet>
+
+</>
+}>
+
+### Create and Run the Worker
+
+With your Activity and Workflow defined, you need a Worker to execute them. A Worker polls a Task Queue, that you configure it to poll, looking for work to do. Once the Worker dequeues the a Workflow or Activity task from the Task Queue, it then executes that task.
+
+Workers are a crucial part of your Temporal application as they're what actually execute the tasks defined in your Workflows and Activities.
+For more information on Workers, see [Understanding Temporal](/evaluate/understanding-temporal#workers) and a [deep dive into Workers](/workers).
+
+Create a Worker file (`worker.ts`):
+
+Keep this terminal running.
+
+</SetupStep>
+
+<SetupStep code={
+<>
+
+<div className="execution-options">
+  <h4>Choose Your Execution Method</h4>
+  <p>With the Worker running, you can start a Workflow in two ways:</p>
+</div>
+
+    <div className="option-section">
+      <h5>Option 1: Using Temporal CLI (Recommended for testing)</h5>
+      <p>Open a new terminal and run:</p>
+      <CodeSnippet language="bash">
+      temporal workflow start --type sayHelloWorkflow --task-queue my-task-queue --input '"Temporal"'
```

Comment:
I keep getting this error when I run this command:
`time=2025-07-17T10:25:35.738 level=ERROR msg="failed reaching server: context deadline exceeded"`

Reply:
This fixed the error: `temporal env set default.address 127.0.0.1:7233`